cmd/soroban-rpc/internal/config/config.go

```diff
@@ -43,31 +40,41 @@ type Config struct {
 	SQLiteDBPath                     string
 	TransactionLedgerRetentionWindow uint32
 
-	// We memoize these, so they bind to viper flags correctly
+	// We memoize these, so they bind to pflags correctly
 	optionsCache *ConfigOptions
-	flagsCache   *support.ConfigOptions
-}
-
-func (cfg *Config) Init(cmd *cobra.Command) error {
-	return cfg.flags().Init(cmd)
+	flagset      *pflag.FlagSet
 }
 
-// Values applied as defaults -> flags/env -> config file
-func (cfg *Config) SetValues() error {
+func (cfg *Config) SetValues(lookupEnv func(string) (string, bool)) error {
+	// We start with the defaults
 	if err := cfg.loadDefaults(); err != nil {
 		return err
 	}
 
-	cfg.Bind()
+	// Then we load from the environment variables and cli flags, to try to find
+	// the config file path
+	if err := cfg.loadEnv(lookupEnv); err != nil {
```

Comment:
what if add ability to load flag from env by a key name such as `config-path` key, then the overlay sequence simplifies down to:

1. load flags
2. if config-path flag present, load config-path
3. load env

rather than doing a second global apply of env flags in the config-path conditional?

Reply:
Yeah, I considered that. Would need to make the config path option extractable or something. Couldn't find a way to do it cleanly with DRY, etc... There's no real downside to loading everything twice since loading it is non-destructive.